File: backend/src/main/java/com/back/ourlog/domain/comment/controller/CommentController.java
package com.back.ourlog.domain.comment.controller;

import com.back.ourlog.domain.comment.dto.CommentRequestDto;
import com.back.ourlog.domain.comment.dto.CommentResponseDto;
import com.back.ourlog.domain.comment.dto.CommentUpdateRequestDto;
import com.back.ourlog.domain.comment.service.CommentService;
import com.back.ourlog.domain.user.entity.User;
import com.back.ourlog.global.common.dto.RsData;
import com.back.ourlog.global.rq.Rq;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/v1/comments")
@Tag(name = "댓글 API")
public class CommentController {
    private final CommentService commentService;
    private final Rq rq;
    @PostMapping
    @Operation(summary = "댓글 등록")
    public ResponseEntity<RsData<CommentResponseDto>> writeComment(@RequestBody @Valid CommentRequestDto req) {
        User user = rq.getCurrentUser();

        CommentResponseDto res = commentService.write(req.getDiaryId(), user, req.getContent());

        return ResponseEntity.status(HttpStatus.CREATED)
                .body(RsData.of("201-1", "댓글이 등록되었습니다.", res));
    }

    @GetMapping("/{diaryId}")
    @Operation(summary = "댓글 조회")
    public ResponseEntity<RsData<List<CommentResponseDto>>> getComments(@PathVariable("diaryId") int diaryId) {
        List<CommentResponseDto> res = commentService.getComments(diaryId);

        return ResponseEntity.status(HttpStatus.OK)
                .body(RsData.of("200-1", "%d번 다이어리 댓글이 조회되었습니다.".formatted(diaryId), res));
    }

    @PutMapping()
    @Operation(summary = "댓글 수정")
    public ResponseEntity<RsData<Void>> updateComment(
            @RequestBody CommentUpdateRequestDto req) {
        User user = rq.getCurrentUser();

        commentService.checkCanUpdate(user, req.getId());

        commentService.update(req.getId(), req.getContent());

        return ResponseEntity.status(HttpStatus.OK)
                .body(RsData.of("200-0"
                        ,"%d번 댓글이 수정되었습니다.".formatted(req.getId())
                        , null));
    }

    @DeleteMapping("/{commentId}")
    @Operation(summary = "댓글 삭제")
    public ResponseEntity<RsData<Void>> deleteComment(
            @PathVariable("commentId") int commentId) {
        User user = rq.getCurrentUser();

        commentService.checkCanDelete(user, commentId);

        commentService.delete(commentId);

        return ResponseEntity.status(HttpStatus.OK)
                .body(RsData.of("200-0"
                        ,"%d번 댓글이 삭제되었습니다.".formatted(commentId)
                        , null));
    }
}

File: backend/src/main/java/com/back/ourlog/domain/comment/controller/CommentController.kt
package com.back.ourlog.domain.comment.controller

import com.back.ourlog.domain.comment.dto.CommentRequestDto
import com.back.ourlog.domain.comment.dto.CommentResponseDto
import com.back.ourlog.domain.comment.dto.CommentUpdateRequestDto
import com.back.ourlog.domain.comment.service.CommentService
import com.back.ourlog.global.common.dto.RsData
import com.back.ourlog.global.common.extension.toSuccessResponse
import com.back.ourlog.global.common.extension.toSuccessResponseWithoutData
import com.back.ourlog.global.rq.Rq
import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.tags.Tag
import jakarta.validation.Valid
import lombok.RequiredArgsConstructor
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/v1/comments")
@Tag(name = "댓글 API")
class CommentController(
    private val commentService: CommentService,
    private val rq: Rq
) {
    @PostMapping
    @Operation(summary = "댓글 등록")
    fun writeComment(@RequestBody @Valid req: CommentRequestDto): ResponseEntity<RsData<CommentResponseDto>> {
        val user = rq.currentUser

        val res = commentService.write(req.diaryId, user, req.content)

        return res.toSuccessResponse("댓글이 등록되었습니다.")
    }

    @GetMapping("/{diaryId}")
    @Operation(summary = "댓글 조회")
    fun getComments(@PathVariable("diaryId") diaryId: Int): ResponseEntity<RsData<List<CommentResponseDto>>> {
        val res = commentService.getComments(diaryId)

        return res.toSuccessResponse("${diaryId}번 다이어리 댓글이 조회되었습니다.")
    }

    @PutMapping
    @Operation(summary = "댓글 수정")
    fun updateComment(
        @RequestBody req: CommentUpdateRequestDto
    ): ResponseEntity<RsData<Nothing>> {
        val user = rq.currentUser

        commentService.checkCanUpdate(user, req.id)

        commentService.update(req.id, req.content)

        return toSuccessResponseWithoutData("${req.id}번 댓글이 수정되었습니다.")
    }

    @DeleteMapping("/{commentId}")
    @Operation(summary = "댓글 삭제")
    fun deleteComment(
        @PathVariable("commentId") commentId: Int
    ): ResponseEntity<RsData<Nothing>> {
        val user = rq.currentUser

        commentService.checkCanDelete(user, commentId)

        commentService.delete(commentId)

        return toSuccessResponseWithoutData("${commentId}번 댓글이 삭제되었습니다.")
    }
}